src/components/menus/ContextMenu/ContextMenu.tsx

```diff
@@ -30,26 +32,22 @@ export interface IContextMenuProps extends Omit<TooltipProps, 'content'> {
 }
 
 const ContextMenu = (props: IContextMenuProps) => {
-	const {
-		className,
-		classNameList,
-		classNameListItem,
-		items,
-		onShow,
-		onHide,
-		noBG,
-		bgOnHover,
-		useClickInsteadOfHover,
-		...otherProps
-	} = props;
+	const { className, classNameList, classNameListItem, items, onShow, onHide, noBG, bgOnHover, id, ...otherProps } =
+		props;
+
+	const menuId = useRef(`${id ?? 'contextMenu'}-${shortid.generate()}`);
```

Comment:
Do we always want to randomly generate a hash?

If I understand this correctly, the way this is written we will always get some sort of random id:
```typescript
<ContextMenu
	id="something"
	items={[
        // ...
    ]) />
// will give <div class="..." id="something-PrzwA7yje">...</div>
<ContextMenu
	items={[
        // ...
    ]) />
// will give <div class="..." id="contextMenu-PrzwA7yje">...</div>
```


But since `id` is a Typescript `HTMLElement` and therefore just a regular HTML attribute -- it makes sense to me that we would want to be able to explicitly set an `id` and fallback to a randomized string if none is passed. The main reason that comes to my mind would be ease of testing, but there's probably other reasons. I imagine something like:


```typescript
<ContextMenu
	id="something"
	items={[
        // ...
    ]) />
// should give <div class="..." id="something">...</div>
<ContextMenu
	items={[
        // ...
    ]) />
// should give <div class="..." id="contextMenu-PrzwA7yje">...</div>
```

What do you think?

Reply:
Yeah whenever I do this I always forget parent components might want to be able to explicitly know the ID of element, specifically for playwright tests, yeah. 

I think I'll do what you said and fallback to a unique hash but default to the passed ID. My main concern is there being multiple context menus with the same ID, but that's more on devs to realize not to do that.

Reply:
Also, we'd pretty much only every have one contextmenu open at a time, because clicking outside of one would close the other